Jenkinsfile

```diff
@@ -0,0 +1,11 @@
+#!groovy
+node {
+  stage 'checkout source and unit test'
+  checkout scm
+  sh 'git rev-parse --short HEAD > commit'
+  def sha1 = readFile('commit').trim()
```

Comment:
My understanding is that each build is done in a separate directory. The commit file should be unique to that directory and shouldn't be overwritten by any other build.